src/Symfony/Bridge/Twig/AppVariable.php

```diff
@@ -17,18 +17,20 @@
 use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
 use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
 use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Contracts\Service\ResetInterface;
 
 /**
  * Exposes some Symfony parameters and services as an "app" global variable.
  *
  * @author Fabien Potencier <fabien@symfony.com>
  */
-class AppVariable
+class AppVariable implements ResetInterface
 {
     private TokenStorageInterface $tokenStorage;
     private RequestStack $requestStack;
     private string $environment;
     private bool $debug;
+    private array $flashBagCache = [];
```

Comment:
The code can be more decoupled if you create a class `CachedFlashBag` that decorates the `FlashBag`. This way it would be easier to test and that this feature could be used in non-twig context if needed.

Reply:
I extracted it into its own class. Good feedback :)